include/diaspora.php

```diff
@@ -2031,7 +2092,7 @@ private function original_item($guid, $orig_author, $author) {
 			if (self::is_reshare($r[0]["body"], true))
 				$r = array();
 			elseif (self::is_reshare($r[0]["body"], false)) {
-				$r[0]["body"] = diaspora2bb(bb2diaspora($r[0]["body"]));
+				$r[0]["body"] = self::replace_people_guid(diaspora2bb(bb2diaspora($r[0]["body"])), $r[0]["author-link"]);
```

Comment:
Generally chaining functions/methods is discouraged as errorous return values are not being handled properly. I would recommend rewriting this and check on valid return types.

Alternatively (maybe for a complete refacturing of the code) throw proper exceptions: http://de.php.net/manual/de/spl.exceptions.php